app/main.py

```diff
@@ -1,3 +1,73 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        elif isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        raise TypeError("Unsupported operand type(s) for +")
+
+    def __iadd__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            self.km += other
+        elif isinstance(other, Distance):
+            self.km += other.km
+        else:
+            raise TypeError("Unsupported operand type(s) for +=")
+        return self
+
+    def __mul__(self, other: Union[int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        raise TypeError("Unsupported operand type(s) for *")
+
+    def __truediv__(self, other: Union[int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
```

Comment:
Consider adding a check to handle division by zero. You can raise a `ZeroDivisionError` if `other` is zero to prevent runtime errors.